client/map_verifier.go

```diff
@@ -86,3 +88,26 @@ func (m *MapVerifier) VerifyMapLeafInclusionHash(rootHash []byte, leafProof *tri
 func (m *MapVerifier) VerifySignedMapRoot(smr *trillian.SignedMapRoot) (*types.MapRootV1, error) {
 	return tcrypto.VerifySignedMapRoot(m.PubKey, m.SigHash, smr)
 }
+
+// VerifyMapLeavesResponse verifies the responses of GetMapLeaves and GetMapLeavesByRevision.
```

Comment:
nit: text describes how the caller might use the function rather than what the function does.   Something like `VerifyMapLeavesResponse verifies the contents of a GetMapLeavesResponse` would not go stale if some future entrypoint re-used the `GetMapLeavesResponse` message.